content/300-guides/200-deployment/110-deployment-guides/550-deploying-to-deno-deploy.mdx

```diff
@@ -0,0 +1,229 @@
+
+</TopBlock>
+
+## Prerequisites
+
+- Free [Deno Deploy](https://deno.com/deploy) account
+- Free [GitHub](https://github.com/) account
+- A PostgreSQL database
+- Node.js & NPM installed
+- Git installed
+
+## 1. Set up your Application
+
+Open your terminal and navigate to a location of your choice. Run the following commands to set up your application.
+
+```terminal
+mkdir prisma-deno-deploy
+cd prisma-deno-deploy
+deno run -A --unstable npm:prisma init
+```
+
+Inside the generated `prisma/schema.prisma`, add the following schema:
+
+```prisma
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model Log {
+  id      String @id @default(autoincrement())
+  level   Level
+  message String
+  meta    Json
+}
+
+enum Level {
+  Info
+  Warn
+  Error
+}
+```
+
+This data model will be used to store incoming requests from your application.
+
+In your `.env` file, replace the current placeholder connection string with your PostgreSQL connection string.
+
+## 2. Create the database schema
+
+With the data model created and your database connection configured, you can now create the data model in your database:
+
+```terminal
+deno run -A --unstable npm:prisma db push
+```
+TODO does this already generate?
+
+This will create the `Log` table and `Level` enum:
+
+```
+TODO
+```
+
+## 3. Generate a Prisma Client
+
+Next, you'll generate a Prisma Client that connects through the [Data Proxy](/data-platform/data-proxy) over HTTP.
+
+```terminal
+npx prisma generate --data-proxy
+```
+
+## 4. Create your application and test it locally
+
+With the database schema created, and a local Prisma Client generated you can now create a minimal local Deno application. Create an index.ts with the following content:
+
+```ts
+import { serve } from "https://deno.land/std@0.140.0/http/server.ts";
+import { PrismaClient } from './generated/client/deno/edge.ts'
+
+const prisma = new PrismaClient({
+    log: ['query', 'info', 'warn', 'error'],
+  })
+
+async function handler(_req: Request) {
+  // TODO Rewrite to write to log
+  const users = await prisma.user.findMany()
+  const body = JSON.stringify(users, null, 2);
+
+/*
+  const { request } = event
+
+  // waitUntil method is used for sending logs, after response is sent
+  event.waitUntil(
+    prisma.log.create({
+      data: {
+        level: 'Info',
+        message: `${request.method} ${request.url}`,
+        meta: {
+          headers: JSON.stringify(request.headers),
+        },
+      },
+    }).then()
+  )
+
+  return new Response(`request method: ${request.method}!`)
+  */
+
+  return new Response(body, {
+    headers: { "content-type": "application/json; charset=utf-8" },
+  });
+}
+
+serve(handler);
+```
+
+You can't execute this script yet, as you are lacking the credentials for Data Proxy that allow you to use this Client with your database. Let's get that fixed.
```

Comment:
```suggestion
You cannot execute this script yet, because you lack the credentials for Data Proxy that let you use this Client with your database. In the next section, you will resolve this.
```